tests/VersionTest.php

```diff
@@ -221,4 +222,25 @@ public function getInvalidVersionStrings() : array
             'leadingZeroIsInvalid' => ['1.05.2'],
         ];
     }
+
+    /**
+     * @test
+     */
+    public function it_can_set_custom_comparator() : void
```

Comment:
I'd name it `it_allows_setting_custom_comparator`, as unlike other test cases, it's not something that can be done _with_ or _to_ the version itself